api.go

```diff
@@ -658,12 +658,8 @@ func (c Client) newRequest(method string, metadata requestMetadata) (req *http.R
 	case c.signature.isV4():
 		// Set sha256 sum for signature calculation only with signature version '4'.
 		shaHeader := unsignedPayload
-		if !c.secure {
-			if metadata.contentSHA256Bytes == nil {
-				shaHeader = hex.EncodeToString(sum256([]byte{}))
-			} else {
-				shaHeader = hex.EncodeToString(metadata.contentSHA256Bytes)
-			}
+		if len(metadata.contentSHA256Bytes) > 0 {
```

Comment:
I was coming from the point that `nil` is a sentinel value for `[]T` and makes it easier to recognise as a pattern while reading code. `if len(contentSHA256Bytes) > 0` allows users to send `[]byte{}` and `nil` interchangeably, which is convenient, so this check is fine by me.